packages/kernel/src/client.ts

```diff
@@ -107,7 +108,69 @@ export class LiteKernelClient implements Kernel.IKernelAPIClient {
             error instanceof Error &&
             error.message.includes('request for lock canceled')
           ) {
-            // expected to throw when mutex.cancel() is called below
+            // expected to throw when mutex.cancel() is called below on cell execution error or on interrupt
+            const cancelReason = this._cancelReason.get(mutex);
+            if (
+              (cancelReason === 'interrupt' ||
+                cancelReason === 'interrupt-subsequent') &&
+              msg.header.msg_type === 'execute_request'
+            ) {
+              if (cancelReason === 'interrupt') {
+                // Change cancel reason so that only one cell includes the error.
+                // Needs to go before await for mutex.
+                this._cancelReason.set(mutex, 'interrupt-subsequent');
+              }
+              await mutex.waitForUnlock();
+              // Send interrupt error to all clients
+              const content: KernelMessage.IReplyErrorContent = {
+                status: 'error',
+                ename: 'Kernel Interrupt',
+                evalue: 'Interrupted',
+                traceback: [],
+              };
+              const sendMessage = this._kernelSends.get(kernelId);
+              if (sendMessage === undefined) {
+                console.warn('Did not find kernel send method');
+                return;
+              }
+              if (cancelReason === 'interrupt') {
+                sendMessage(
+                  KernelMessage.createMessage<KernelMessage.IErrorMsg>({
+                    channel: 'iopub',
+                    session: clientId,
+                    parentHeader: msg.header,
+                    msgType: 'error',
+                    content,
+                  }),
+                );
+              }
+              sendMessage(
+                KernelMessage.createMessage<KernelMessage.IExecuteReplyMsg>({
+                  channel: 'shell',
+                  session: clientId,
+                  parentHeader: (msg as KernelMessage.IExecuteRequestMsg).header,
+                  msgType: 'execute_reply',
+                  content: {
+                    ...content,
+                    execution_count: 0,
+                  },
+                  metadata: {
+                    casuse: 'interrupt',
```

Comment:
Should this be named `cause`?

```suggestion
                    cause: 'interrupt',
```